lib/src/components/NxSkeletonLoader/NxSkeletonLoader.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { HTMLAttributes, ReactNode, useEffect } from 'react';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { find } from 'ramda';
+import withClass from '../../util/withClass';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import SkeletonContext from './SkeletonContext';
+
+import './NxSkeletonLoader.scss';
+
+export { SkeletonContext };
+
+const ANIMATION_NAME = 'skeleton-animation';
+
+// These two variables help globally synchronize skeleton loader animation state. Whenever a skeleton loader
+// is present on the page, any additional skeleton loaders should synchronize with it. Once all skeleton loaders
+// are removed, that synchronization state should be cleared so that a future skeleton loader added later will
+// start its animation cleanly at the beginning and not based on the original skeleton loader's time
+const activeSkeletonLoaders = new Set();
+let animationStartTime: number | null = null;
+
+/*
+ * According to the spec, animations have an id property (that is presumably supposed
+ * to be the CSS keyframe name) and not an animationName property. However, at the time of writing,
+ * in at least Chrome and firefox the id for our animation is the empty string and an `animationName` property
+ * contains the name of the CSS keyframes
+ */
+function isSkeletonAnimation(animation: Animation) {
+  return animation.id === ANIMATION_NAME || (animation as { animationName?: string }).animationName === ANIMATION_NAME;
+}
+
+function _NxSkeletonLoader({ children }: { children: ReactNode }) {
+  /*
+   * Synchronize all skeleton animations
+   */
+  useEffect(() => {
+    // a unique object to identify this NxSkeletonLoader instance in the activeSkeletonLoaders set
+    const instance = {};
+
+    function handleAnimationStart({ animationName, target }: AnimationEvent) {
+      if (animationName === ANIMATION_NAME) {
+        const animation = target instanceof Element && find(isSkeletonAnimation, target.getAnimations());
+
+        if (animation) {
+          if (animationStartTime) {
+            animation.startTime = animationStartTime;
+          }
+          else {
+            animationStartTime = animation.startTime;
+          }
+        }
+      }
+    }
+
+    if (!activeSkeletonLoaders.size) {
+      window.addEventListener('animationstart', handleAnimationStart);
+    }
+
+    activeSkeletonLoaders.add(instance);
```

Comment:
I'm having a bit of trouble trying to understand the use of `instance`- why are we passing an empty object to the activeSkeletonLoaders Set instead of the target or animation?

Reply:
> I'm having a bit of trouble trying to understand the use of instance- why are we passing an empty object to the activeSkeletonLoaders Set instead of the target or animation?

The empty object is a stand-in for the `NxSkeletonLoader` component instance. Think of `activeSkeletonLoaders` as a set of currently-mounted `NxSkeletonLoader` react component instances.  Except, there is no "react component instance" object that we have access to that we could store in the set.  So instead, each one creates an object that we do have access to and stores it in the set as a stand-in.  The only data carried by the objects is their identity - none of the objects are considered to be equal to one another by the set.  

Thinking about it more,  I suppose the set could be replaced by a simple counter that gets incremented in each `useEffect` and decremented in each `useEffect` unregistration callback.  All we need to know is whether there are currently zero or non-zero `NxSkeletonLoaders` mounted.  I'll change it to that.

Reply:
>The empty object is a stand-in for the NxSkeletonLoader component instance. Think of activeSkeletonLoaders as a set of currently-mounted NxSkeletonLoader react component instances. Except, there is no "react component instance" object that we have access to that we could store in the set. So instead, each one creates an object that we do have access to and stores it in the set as a stand-in. The only data carried by the objects is their identity - none of the objects are considered to be equal to one another by the set.

Thanks for the clarification!